File: web2-Project/src/app/entities/flightCompany/flight-company.ts
import { Address } from '../address/address';
import { Flight } from '../flight/flight'
import { AvioSediste } from '../avio-sediste/avio-sediste'
import { FlightDestination } from '../flight-destination/flight-destination';
import { User } from '../user/user';

export class FlightCompany {
    id: number; // jedinstevi identifikacioni broj kompanije
    name: string;
    addressId: number;
    address: Address; // adresa gde se aerodrom nalazi
    promotionalDesc: string; // promotivni opis
    destinations: Array<FlightDestination>; // adrese na koje lete avioni kompanije
    flights: Array<Flight>; // svi trenutni letovi kompanije
    logo: string;
    ocene: Array<number>;

    adminId: string;
    admin: User;

    deleted: boolean;

    constructor(
        id: number,
        name: string, 
        address: Address, 
        promotionalDesc: string,
        destinations: Array<FlightDestination>,
        flights: Array<Flight>,
        logo: string,
        ocene: Array<number>) {
            this.id = id;
            this.name = name;
            this.address = address;
            this.promotionalDesc = promotionalDesc;
            this.destinations = destinations;
            this.flights = flights;
            this.logo = logo;
            this.ocene = ocene;

            this.deleted = false;
    }
}

File: MAANPP20/MAANPP20/Common/Models/Flights/FlightCompany.cs
using Common.Models.Common;
using Common.Models.Common_U;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Common.Models.Flights
{
    public class FlightCompany
    {
        [Key]
        public int id { get; set; }

        [Required]
        [StringLength(25)]
        public string name { get; set; }

        [ForeignKey("address")]
        public int addressId { get; set; }
        public Address address { get; set; }

        [Required]
        public string promotionalDesc { get; set; }

        public ICollection<FlightDestination> destinations { get; set; }

        public ICollection<Flight> flights { get; set; }

        public string logo { get; set; }

        public ICollection<DoubleForICollection> ocene { get; set; }

        [ForeignKey("admin")]
        public string idAdmin { get; set; }
        public User admin { get; set; }

        public bool deleted { get; set; }
    }
}
